source/analysis/AM2DAdditionAB.h

```diff
@@ -77,13 +80,15 @@ protected slots:
 	/// Helper function to look at our overall situation and determine what the output state should be.
 	void reviewState();
 
-    /// Computes the cached data for access getters value() and values().
-    void computeCachedValues() const;
+	/// Computes the cached data for access getters value() and values().
+	void computeCachedValues() const;
 
-    /// Flag for knowing whether we need to compute the values.
-    mutable bool cacheUpdateRequired_;
-    /// The vector holding the data.
-    mutable QVector<double> cachedData_;
+	/// Flag for knowing whether we need to compute the values.
+	mutable bool cacheUpdateRequired_;
+	/// The vector holding the data.
+	mutable QVector<double> cachedData_;
+	/// Holds the cached data range.
+	mutable AMRange cachedDataRange_;
```

Comment:
Yes, they certainly could.  Unfortunately, that involves making a database upgrade since every single analysis block would need to be redone.  I'd love to do some AMDataSource/AMAnalysisBlock refactoring, but at the moment it seems like a rather large undertaking.